cmd/image/qcow2ova/gzip.go

```diff
@@ -0,0 +1,77 @@
+package qcow2ova
+
+import (
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	gzip "github.com/klauspost/pgzip"
+)
+
+// gzip the source file to target
+func gzipit(source, target string) error {
+	reader, err := os.Open(source)
+	if err != nil {
+		return err
+	}
+
+	filename := filepath.Base(source)
+	writer, err := os.Create(target)
+	if err != nil {
+		return err
+	}
+	defer writer.Close()
+
+	archiver := gzip.NewWriter(writer)
+	defer archiver.Close()
+	archiver.Name = filename
+
+	_, err = io.Copy(archiver, reader)
+	return err
+}
+
+// gunzip the source file to target
+func gunzip(source, target string) error {
+	reader, err := os.Open(source)
+	if err != nil {
+		return err
+	}
+	defer reader.Close()
+
+	archive, err := gzip.NewReader(reader)
+	if err != nil {
+		return err
+	}
+	defer archive.Close()
+
+	writer, err := os.Create(target)
+	if err != nil {
+		return err
+	}
+	defer writer.Close()
+
+	_, err = io.Copy(writer, archive)
+	return err
+}
+
+// isGzip returns if file is in gzip format
+func isGzip(source string) (bool, error) {
+	file, err := os.Open(source)
+	if err != nil {
+		return false, err
+	}
+	defer file.Close()
+
+	buff := make([]byte, 512)
+	_, err = file.Read(buff)
+	if err != nil {
+		return false, err
+	}
+
+	if filetype := http.DetectContentType(buff); filetype == "application/x-gzip" {
```

Comment:
So content type detection is only via http module? any other alternatives ?

Reply:
To clarify the above comment, is there some magic string that we can search in buff ?

Reply:
I tried looking for the existing standard library, I didn't find anything useful other than this `net/http` module, infact this module also got the same logic what you mentioned like checking the 512 magic bytes from the file.